### DISCUSSION TITLE
Flows: Upload JSON File ?

### DISCUSSION THREAD

**Discussion by MentalGear** [directus/directus#16777] [Archive]
I generate a JSON file in a Flow which I would like upload to Directus (Cloud).

First I tried to use the `Run Script`, which I can't use since it doesn't have FormData (no node.js).
Then, I tried with the `Request Node`, which has no option in the body to send FormData it seems.

Now I'm wondering if uploading a simple JSON file to Directus via Flows is even possible.

**Comment by br41nslug** [comment]
In the cloud you currently cannot allow the required packages to upload a file from within the run script sandbox. You'll have to save your generated JSON data to a collection in the database

**Reply by MentalGear** [reply]
Thx for the reply. Unfortunately I need it as a static file. Is there no way, maybe using the `Webhook` node ?

**Comment by paescuj** [comment]
> First I tried to use the Run Script, which I can't use since it doesn't have FormData (no node.js).

~What do you mean by this? `FormData` isn't specific to Node.js: https://developer.mozilla.org/en-US/docs/Web/API/FormData~

Ah sorry, the "(no node.js)" confused me, you meant that it is not available in Node.js... I've just submitted a pull request to update the Docker image to Node.js v18 where `FormData` will be available: #16810

**Reply by fitu86** [reply]
Look, this is how I import the file
![Screenshot_20221215_180021](https://user-images.githubusercontent.com/16139127/207992180-e8561af3-26b7-4e86-8988-cc5f9a56d411.jpg)
s from flow

**Reply by MentalGear** [reply]
Thx for your approach and especially the pull request!
I also started developing a method where I use pipedream (integration service) to first send the JSON payload to them, and have them upload it as a file to Directus, which is quite the detour and having everything in directus would be great.
So, with FormData this should be possible (having an Upload action in Flows also wouldn't hurt).

**Reply by ericandrewscott** [reply]
@fitu86 
I actually did the same exact thing, but I get:
```
{
  "message": "Cannot read properties of undefined (reading 'status')"
}
```
And can't figure out how to not get this error. I can hit the same endpoint with the bearer and same payload with Postman. 
I've also removed `0.0.0.0` from the env var `IMPORT_IP_DENY_LIST`.

**Reply by fitu86** [reply]
try these values in the .env of the target application.

HOST="127.0.0.1"
PUBLIC_URL="https://example.com/"
IMPORT_IP_DENY_LIST=""


Are you using nginx?

**Reply by ericandrewscott** [reply]
No. Docker + Redis.
I'll give those a try. 
Should `IMPORT_IP_DENY_LIST` be an empty array `[]` or an empty string?

**Reply by fitu86** [reply]
empty string

**Reply by ericandrewscott** [reply]
For some reason, that environment variable for `HOST` causes my deployment to fail. And without it, I'm still getting the error from earlier in the thread 😢
